graphyte.py

```diff
@@ -64,31 +64,39 @@ def stop(self):
             self._thread.join()
             self.interval = None
 
-    def build_message(self, metric, value, timestamp):
+    def build_message(self, metric, value, timestamp, tags={}):
         """Build a Graphite message to send and return it as a byte string."""
         if not metric or metric.split(None, 1)[0] != metric:
             raise ValueError('"metric" must not have whitespace in it')
         if not isinstance(value, (int, float)):
             raise TypeError('"value" must be an int or a float, not a {}'.format(
-                    type(value).__name__))
+                type(value).__name__))
+        if not isinstance(tags, dict):
+            raise TypeError('"tags" must be a dict, not a {}'.format(
+                type(tags).__name__))
 
-        message = u'{}{} {} {}\n'.format(
+        tags_suffix = ''.join(';{}={}'.format(x[0], x[1]) for x in sorted(tags.items()))
```

Comment:
Looks good -- nice idea to sort the order of the tags to produce consistent output and make testing easier.